tests/v2/validation/provisioning/rke1/etcd_node_delete_test.go

```diff
@@ -0,0 +1,158 @@
+	suite.Suite
+	session                *session.Session
+	client                 *rancher.Client
+	ns                     string
+	rke1kubernetesVersions []string
+	cnis                   []string
+	providers              []string
+	nodesAndRoles          []nodepools.NodeRoles
+}
+
+func (e *EtcdNodeDeleteTestSuite) TearDownSuite() {
+	e.session.Cleanup()
+}
+
+func (e *EtcdNodeDeleteTestSuite) SetupSuite() {
+	testSession := session.NewSession()
+	e.session = testSession
+
+	e.ns = defaultNS
+
+	clustersConfig := new(provisioning.Config)
+	config.LoadConfig(provisioning.ConfigurationFileKey, clustersConfig)
+
+	e.rke1kubernetesVersions = clustersConfig.RKE1KubernetesVersions
+
+	e.cnis = clustersConfig.CNIs
+	e.providers = clustersConfig.Providers
+	e.nodesAndRoles = clustersConfig.NodesAndRolesRKE1
+
+	client, err := rancher.NewClient("", testSession)
+	require.NoError(e.T(), err)
+
+	e.client = client
+}
+
+func (e *EtcdNodeDeleteTestSuite) isEtcdNodeRemoved(etcdNodeToDelete management.Node, clusterResp *management.Cluster) (int, error) {
+	noOfEtcdNodesAfterDeletion := 0
+
+	err := kwait.Poll(5*time.Second, 15*time.Minute, func() (done bool, err error) {
+		machines, err := e.client.Management.Node.List(&types.ListOpts{Filters: map[string]interface{}{
+			"clusterId": clusterResp.ID,
+		}})
+		if err != nil {
+			return false, err
+		}
+		noOfEtcdNodesAfterDeletion = 0
+		for _, machine := range machines.Data {
+			if machine.Etcd {
+				if machine.ID == etcdNodeToDelete.ID {
+					return false, nil
+				}
+				noOfEtcdNodesAfterDeletion++
+			}
+		}
+		logrus.Info("new etcd node : ")
+		for _, machine := range machines.Data {
+			if machine.Etcd {
+				logrus.Info(machine.NodeName)
+			}
+		}
+		return true, nil
+	})
+	return noOfEtcdNodesAfterDeletion, err
+}
+
+func (e *EtcdNodeDeleteTestSuite) TestEtcdDeleteNode() {
+	require.GreaterOrEqual(e.T(), len(e.providers), 1)
+	require.GreaterOrEqual(e.T(), len(e.cnis), 1)
+	require.GreaterOrEqual(e.T(), len(e.rke1kubernetesVersions), 1)
+
+	subSession := e.session.NewSession()
+	defer subSession.Cleanup()
+
+	client, err := e.client.WithSession(subSession)
+	require.NoError(e.T(), err)
+
+	for _, provider := range e.providers {
+		for _, k8sVersion := range e.rke1kubernetesVersions {
+			for _, cni := range e.cnis {
+				provider := CreateProvider(provider)
+				nodeTemplate, err := provider.NodeTemplateFunc(client)
+				require.NoError(e.T(), err)
+
+				logrus.Info("provisioning ", k8sVersion, " cluster using ", cni, " CNI..")
+				clusterResp, err := TestProvisioningRKE1Cluster(e.T(), client, provider, e.nodesAndRoles, k8sVersion, cni, nodeTemplate)
+				require.NoError(e.T(), err)
+
+				machines, err := e.client.Management.Node.List(&types.ListOpts{Filters: map[string]interface{}{
+					"clusterId": clusterResp.ID,
+				}})
+				require.NoError(e.T(), err)
+
+				noOfEtcdNodesBeforeDeletion := 0
+				etcdNodeToDelete := management.Node{}
+
+				logrus.Info("etcd nodes before deletion : ")
```

Comment:
Same comment as before, I do not have a strong preference either way. But I think this logging can just be removed as you're already logging the etcd node to be removed.

Reply:
Removed it.